File: nexus-api/src/main/java/com/nexus/platform/service/UserService.java
package com.nexus.platform.service;

import com.nexus.platform.dto.user.UserRequest;
import com.nexus.platform.dto.user.UserPermissionsResponse;
import com.nexus.platform.dto.user.UserResponse;
import com.nexus.domain.enums.UserStatus;
import com.nexus.platform.dto.user.PasswordUpdateRequest;
import com.nexus.platform.dto.user.ProfileUpdateRequest;

import java.util.List;

public interface UserService {

    List<UserResponse> findAll(String currentUserEmail);

    UserResponse findById(Long id, String currentUserEmail);

    UserResponse findByEmail(String email, String currentUserEmail);

    UserResponse currentUser(String email);

    UserPermissionsResponse currentUserPermissions(String email);

    UserResponse updateProfile(String email, ProfileUpdateRequest request);

    void updatePassword(String email, PasswordUpdateRequest request);

    UserResponse create(UserRequest request);

    UserResponse create(UserRequest request, String currentUserEmail);

    UserResponse update(Long id, UserRequest request);

    UserResponse update(Long id, UserRequest request, String currentUserEmail);

    UserResponse updateStatus(Long id, UserStatus status, String currentUserEmail);

    void delete(Long id, String currentUserEmail);
}

File: frontend/src/services/userService.ts
import { apiClient } from "@/lib/api";
import type { PasswordPayload, ProfilePayload, User, UserPayload, UserStatus } from "@/types/user";

const USERS_ENDPOINT = "/api/users";

export const userService = {
  async findAll() {
    const response = await apiClient.get<User[]>(USERS_ENDPOINT);
    return response.data;
  },

  async findById(id: number) {
    const response = await apiClient.get<User>(`${USERS_ENDPOINT}/${id}`);
    return response.data;
  },

  async currentUser() {
    const response = await apiClient.get<User>(`${USERS_ENDPOINT}/me`);
    return response.data;
  },

  async updateProfile(payload: ProfilePayload) {
    const response = await apiClient.put<User>(`${USERS_ENDPOINT}/me`, payload);
    return response.data;
  },

  async updatePassword(payload: PasswordPayload) {
    await apiClient.put(`${USERS_ENDPOINT}/me/password`, payload);
  },

  async create(payload: UserPayload) {
    const response = await apiClient.post<User>(USERS_ENDPOINT, payload);
    return response.data;
  },

  async update(id: number, payload: UserPayload) {
    const response = await apiClient.put<User>(`${USERS_ENDPOINT}/${id}`, payload);
    return response.data;
  },

  async updateStatus(id: number, status: UserStatus) {
    const response = await apiClient.patch<User>(`${USERS_ENDPOINT}/${id}/status`, { status });
    return response.data;
  },

  async remove(id: number) {
    await apiClient.delete(`${USERS_ENDPOINT}/${id}`);
  },
};
